plugins/inputs/docker/docker.go

```diff
@@ -565,18 +599,60 @@ func (d *Docker) gatherContainerInspect(
 		}
 	}
 
-	parseContainerStats(v, acc, tags, container.ID, d.PerDevice, d.Total, daemonOSType)
+	// Temporary logic needed for backwards compatibility until 'perdevice' setting is removed.
+	perDeviceInclude := getEffectivePerDeviceInclude(d.PerDevice, d.PerDeviceInclude, d.Log)
+	// Temporary logic needed for backwards compatibility until 'total' setting is removed.
+	totalInclude := getEffectiveTotalInclude(d.Total, d.TotalInclude, d.Log)
```

Comment:
Please do it once in the `Init()` function. This way `Gather()` is clean wrt backward-case handling.

Reply:
Indeed! Didn't realize I was polluting `Gather()` which is not suited for a one time logic